packages/node-cli/src/core/sdksetup/SdkDownloadService.js

```diff
@@ -41,52 +45,73 @@ class SdkDownloadService {
-			})
+			});
 			NodeConsoleLogger.info(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_SUCCESS));
-		}
-		catch (error) {
-			NodeConsoleLogger.error(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_ERROR, fullURL, unwrapExceptionMessage(error)))
+		} catch (error) {
+			NodeConsoleLogger.error(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_ERROR, fullURL, unwrapExceptionMessage(error)));
 		}
 	}
 
-	_downloadFile(url, sdkDirectory) {
-		const proxy = process.env.npm_config_https_proxy || process.env.npm_config_proxy;
-		const isProxyRequired = proxy && !SdkProperties.configFileExists();
-		const removeJarFilesFrom = this._removeJarFilesFrom;
+	downloadJarFilePromise(downloadUrl, destinationFilePath, proxy, skipProxy) {
+		const isProxyRequired = proxy && !skipProxy;
+		const downloadUrlObject = new URL(downloadUrl);
+		const downloadUrlProtocol = downloadUrlObject.protocol;
 
-		const options = {
-			method: 'GET',
-			uri: url,
+		const requestOptions = {
 			encoding: 'binary',
-			resolveWithFullResponse: true,
-			...(isProxyRequired && { proxy: proxy }),
+			...(isProxyRequired && { agent: new ProxyAgent(proxy, { tunnel: true, timeout: 15000 }) }),
 		};
 
-		return request(options).then(function(response) {
-			if (!VALID_JAR_CONTENT_TYPES.includes(response.headers['content-type'])) {
-				throw NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_ERROR_FILE_NOT_AVAILABLE);
-			}
-
-			// remove all jar files before writing response to file
-			removeJarFilesFrom(sdkDirectory)
+		if (!/^http:$|^https:$/.test(downloadUrlProtocol)) {
+			throw new Error('Invalid SDK jar file download url protocol. Only http: or https: are allowed.');
+		}
 
-			const sdkDestinationFile = path.join(sdkDirectory, SdkProperties.getSdkFileName());
-			const file = fs.createWriteStream(sdkDestinationFile);
-			file.write(response.body, 'binary');
-			file.end();
+		const httpx = 'http:'.match(downloadUrlObject.protocol) ? http : https;
+
+		return new Promise((resolve, reject) => {
+			const clientReq = httpx.get(downloadUrlObject, requestOptions, (response) => {
+				const chunks = [];
+				response.on('data', (chunk) => chunks.push(Buffer.from(chunk, 'binary')));
+
+				response.on('end', () => {
+					if (!VALID_JAR_CONTENT_TYPES.includes(response.headers['content-type'])) {
+						reject(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_ERROR_FILE_NOT_AVAILABLE));
+					}
+
+					const jarFile = fs.createWriteStream(destinationFilePath);
+					jarFile.write(Buffer.concat(chunks), 'binary');
+					jarFile.end();
+					resolve();
+				});
+
+				response.on('error', (error) => {
+					reject(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_ERROR_FILE_NOT_AVAILABLE));
+				});
+			});
+
+			clientReq
+				.once('timeout', () => {
+					clientReq.destroy();
+					reject(new Error(`GET request timeout.`));
+				})
+				.once('error', (err) => reject(err))
+				.end();
```

Comment:
What do you think about moving this whole Promise to a new method?

Reply:
Not sure about it, because this promise is using 4 variables of the context (httpx, downloadUrlObject, requestOptions and destinationFilePath) and we will need to forward them as arguments. But everything can be done.